illumio_connector.py

```diff
@@ -1,249 +1,341 @@
+
+        if self.connect_pce():
+            self.save_progress("Connectivity Test Passed")
+            return action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            self.save_progress("Connectivity Test Failed")
+            return action_result.set_status(phantom.APP_ERROR)
 
-        return RetVal(
-            action_result.set_status(
-                phantom.APP_ERROR, "Empty response and no information in the header"
-            ), None
+    def _handle_get_traffic_analysis(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        ret_val, end_time = self.convert_to_iso(
+            param["end_time"], action_result, "end_time"
         )
 
-    def _process_html_response(self, response, action_result):
-        # An html response, treat it like an error
-        status_code = response.status_code
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
 
-        try:
-            soup = BeautifulSoup(response.text, "html.parser")
-            error_text = soup.text
-            split_lines = error_text.split('\n')
-            split_lines = [x.strip() for x in split_lines if x.strip()]
-            error_text = '\n'.join(split_lines)
-        except:
-            error_text = "Cannot parse error details"
+        ret_val, start_time = self.convert_to_iso(
+            param["start_time"], action_result, "start_time"
+        )
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
 
-        message = "Status Code: {0}. Data from server:\n{1}\n".format(status_code, error_text)
+        if self.check_starttime_greater_than_endtime(start_time, end_time):
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "The 'end_time' parameter must be greater than 'start_time' parameter",
+            )
 
-        message = message.replace(u'{', '{{').replace(u'}', '}}')
-        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+        port = self._validate_integers(self, param["port"], "port")
+        if self._port is None:
+            return self.get_status()
 
-    def _process_json_response(self, r, action_result):
-        # Try a json parse
-        try:
-            resp_json = r.json()
-        except Exception as e:
-            return RetVal(
-                action_result.set_status(
-                    phantom.APP_ERROR, "Unable to parse JSON response. Error: {0}".format(str(e))
-                ), None
+        if port <= 0 or port >= 65536:
+            return action_result.set_status(
+                phantom.APP_ERROR, "Please enter a valid value for 'port' parameter"
             )
 
-        # Please specify the status codes here
-        if 200 <= r.status_code < 399:
-            return RetVal(phantom.APP_SUCCESS, resp_json)
+        protocol = param["protocol"].lower()
+        if protocol not in ["tcp", "udp"]:
```

Comment:
Would be better to move these proto string values to `illumio_consts.py` as an enum or a set.

Reply:
Done